File: test-support/src/main/java/com/rafaelfiume/prosciutto/test/SalumeApiContractExampleReader.java
package com.rafaelfiume.prosciutto.test;

import org.apache.commons.io.IOUtils;

import java.io.IOException;
import java.io.InputStream;

public class SalumeApiContractExampleReader {

    public static String supplierAdviceForExpertRequest() {
        return readFile("AdviseProductBasedOnCustomerProfileEndToEndTest/request.onlySuggestTraditionalProductsWithCheapestOnesFirstToExperts.txt");
    }

    public static String supplierAdviceForExpertResponse() {
        return readFile("AdviseProductBasedOnCustomerProfileEndToEndTest/response.onlySuggestTraditionalProductsWithCheapestOnesFirstToExperts.txt");
    }

    public static String readFile(String resName) {
        try (final InputStream is = SalumeApiContractExampleReader.class.getClassLoader().getResourceAsStream(resName)) {
            return IOUtils.toString(is);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: test-support/src/main/kotlin/com/rafaelfiume/prosciutto/test/SalumeApiContractExampleReader.kt
package com.rafaelfiume.prosciutto.test

import org.apache.commons.io.IOUtils
import java.io.IOException

object SalumeApiContractExampleReader {

    fun supplierAdviceForExpertRequest(): String = readFile("AdviseProductBasedOnCustomerProfileEndToEndTest/request.onlySuggestTraditionalProductsWithCheapestOnesFirstToExperts.txt")

    fun supplierAdviceForExpertResponse(): String = readFile("AdviseProductBasedOnCustomerProfileEndToEndTest/response.onlySuggestTraditionalProductsWithCheapestOnesFirstToExperts.txt")

    fun readFile(resName: String): String {
        try {
            SalumeApiContractExampleReader::class.java.classLoader
                    .getResourceAsStream(resName).use { `is` -> return IOUtils.toString(`is`) }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }

    }
}
